ambari-server/src/main/java/org/apache/ambari/server/stack/UpdateActiveRepoVersionOnStartup.java

```diff
@@ -88,10 +88,15 @@ public void process() throws AmbariException {
 
           StackId stackId = repositoryVersion.getStackId();
           StackInfo stack = stackManager.getStack(stackId.getStackName(), stackId.getStackVersion());
-
-          if (updateRepoVersion(stack, repositoryVersion)) {
-            repositoryVersionDao.merge(repositoryVersion);
-          }
+        
+          if(stack !=null) {
+            if (updateRepoVersion(stack, repositoryVersion)) {
+              repositoryVersionDao.merge(repositoryVersion);
+            }
+          } else {
+            LOG.error(String.format("Check if Stack %s  version %s is present in file system",  stackId.getStackName(), stackId.getStackVersion()));
+            throw new AmbariException(String.format("Stack %s version %s  was not found in file system",  stackId.getStackName(), stackId.getStackVersion()));
```

Comment:
Change this to: 
`Stack %s was not found on the file system. In the event that it was removed, please ensure that it exists before starting Ambari Server.`

Reply:
> Change this to:
> `Stack %s was not found on the file system. In the event that it was removed, please ensure that it exists before starting Ambari Server.`

Updated.